### PR TITLE
[#4807] remove py3 checks.

### LINKED ISSUE #4807: Remove all remaining Pylons code in core
This should be done in separate pull requests to make reviews easier

- [x] Remove `requirements.py2.*` files and update documentation to remove py2 mentions
- [x] Remove py2 specific code. Look for `if six.PY2:` and remove what's inside!
- [x] Remove py3 checks. Look for `six.PY3` and make that the standard code run (remove the check if any)
- [ ] Remove usage of six. We should not need the compatibility layer any more. Make all code standard Python 3

This should definitely be done separately

- [ ] Remove unicode literals (eg `my_conf[u"test_key_1"] = u"Test value 1"` -> `my_conf["test_key_1"] = "Test value 1"`

### PR DESCRIPTION
**Pull request by steveoni**
remove py3 checks.

Fixes #4807 

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Please [X] all the boxes above that apply

### CODE REVIEW

**Comment by steveoni** [comment]
@amercader pls can you review this #6101 also , so I can update both at the same time.

The test is failing base on javascript/HTML issue, and in this PR I make no changes to any javascript file. I don't know if you have an insight into why this is failing?

**Comment by steveoni** [comment]
@amercader

**Comment by amercader** [comment]
@steveoni this looks good to go but can I ask two small things?

* Can we remove the [`is_flask_request`](https://github.com/ckan/ckan/blob/70420cafa7bc43af01059f2b88beeb45ea8bfff1/ckan/common.py#L26) function and all the places where it is called? Actually we can keep it returning `True` just in case someone was using it from extensions:

```python
@maintain.deprecated('All requests are Flask request', since="2.10.0")
def is_flask_request():
    return True
```

But remove its checks from all these modules:
```python
ckan/ckan/lib/i18n.py
53:from ckan.common import config, is_flask_request, aslist
285:    if is_flask_request():

ckan/ckan/lib/base.py
31:                         session, is_flask_request, asbool)
65:    if is_flask_request():
145:    if not is_flask_request():

ckan/ckan/lib/helpers.py
25:from ckan.common import asbool, config, is_flask_request
234:    if is_flask_request():
805:    if is_flask_request():
939:    if is_flask_request():
1249:            if is_flask_request() else request.params.items()
1287:            if is_flask_request() else request.params.items()
1680:    if is_flask_request():
2117:        if is_flask_request() else request.params.items()
2156:        if is_flask_request() else request.params.items()

ckan/ckan/common.py
31:def is_flask_request():
52:    if is_flask_request():
75:    if is_flask_request():
155:    if is_flask_request():
169:    `is_flask_request`) and at the same time provide all objects methods to be
186:    if is_flask_request():
193:    if is_flask_request():

```
Does this make sense?

* Can you please provide more descriptive commit titles? eg instead of `# changes/4807.removal` use `[#4807] Remove py3 checks` or `[#4807] Remove is_flask_request function`. This makes everybody's lives much easier when reviewing the history or building the changelog 

Thanks for your help!

**Comment by steveoni** [comment]
@amercader sure. i will do that

**Comment by steveoni** [comment]
@amercader

File: ckan/common.py [hunk 1]
```diff
@@ -19,14 +19,16 @@
...
+@maintain.deprecated('All requests are Flask request', since="2.10.0")
 def is_flask_request():
     u'''
-    A centralized way to determine whether we are in the context of a
-    request being served by Flask or Pylons
+    This function is deprecated. There is know use for it
+    since ckan is now fully python 3.* dependent.
     '''
     if six.PY3:
         return True
```
Comment (amercader) [inline ckan/common.py:34]:
```suggestion
    This function is deprecated. All CKAN requests are now served by Flask
    '''
    return True
```

File: ckan/common.py [hunk 2]
```diff
@@ -19,14 +19,16 @@
                          ngettext as flask_ungettext)
 
 import simplejson as json
+import ckan.lib.maintain as maintain
 
 current_app = flask.current_app
 
 
+@maintain.deprecated('All requests are Flask request', since="2.10.0")
```
Comment (amercader) [inline ckan/common.py:27]:
```suggestion
@maintain.deprecated('All web requests are served by Flask', since="2.10.0")
```

**Comment by amercader** [comment]
@steveoni two minor tweaks and we are good to go

**Comment by steveoni** [comment]
@amercader done

**Comment by amercader** [comment]
Thanks @steveoni !

**Pull request merged**